src/lib/steps/rollup.ts

```diff
@@ -22,38 +21,32 @@ export interface RollupOptions {
  * @param opts
  */
 export async function rollup(opts: RollupOptions): Promise<void> {
-  log.debug(`rollup (v${ROLLUP_VERSION}) ${opts.entry} to ${opts.dest} (${opts.format})`);
-
-  const globals = {
-    // default externals for '@angular/*' and 'rxjs'
-    ...ROLLUP_GLOBALS,
-    // external symbols passed from the user's ng-package.json
-    ...opts.externals,
+  log.debug(`rollup (v${__rollup.VERSION}) ${opts.entry} to ${opts.dest} (${opts.format})`);
+  const embedded: string[] = opts.embedded || [];
+  const namedExternals = {
+    'tslib': 'tslib',
+    ...opts.namedExternals,
   };
-  const globalsKeys = Object.keys(globals);
 
   // Create the bundle
   const bundle: __rollup.Bundle = await __rollup.rollup({
     context: 'this',
-    external: (moduleId) => {
-      const isExplicitExternal = globalsKeys.some((global) => global === moduleId);
-      if (isExplicitExternal === true) {
-        return true;
-      }
-
-      // Determine from the path
-      if (moduleId.startsWith('/')) {
-        const moduleIdPath = path.parse(moduleId);
-        const entryPath = path.parse(opts.entry);
+    external: moduleId => {
+      // more information about why we don't check for 'node_modules' path
+      // https://github.com/rollup/rollup-plugin-node-resolve/issues/110#issuecomment-350353632
 
-        // `moduleId` is a file in the sub-tree of `opts.entry` -> inline to bundle
-        if (moduleIdPath.dir.startsWith(entryPath.dir)) {
-          return false;
-        }
+      if (
+        path.isAbsolute(moduleId) ||
+        moduleId.startsWith(".") ||
+        moduleId.startsWith("/") ||
+        moduleId.indexOf("commonjsHelpers") >= 0 || // in case we are embedding a commonjs module we need to include it's helpers also
+        embedded.some(x => x === moduleId)
+      ) {
+        // if it's either 'absolute', marked to embed, starts with a '.' or '/' it's not external
+        return false;
```

Comment:
What do you think of extracting to a function / lambda?

```ts
const externalModuleIdStrategy = (moduleId: string): boolean => {
  /* ... */
}
```

Then write a unit test for it?

Reply:
I’d love that, if you tell me where to place the unit test etc... :) I’d find it much more easy to test.

Reply:
Small correction from your example, to test it, you need to export it though!

Reply:
Ah, yes! 😄